game-core/src/test/java/games/strategy/triplea/ai/BattleCalculatorComparisonTests.java

```diff
@@ -0,0 +1,405 @@
+            territory,
+            attackingUnits,
+            defendingUnits,
+            bombardingUnits,
+            territoryEffects,
+            false,
+            2000);
+
+    printResult(
+        "Hard AI 2000 runs", hardAi2Results, attackingUnits, defendingUnits, territory, data);
+
+    final ProData proData = new ProData();
+    proData.initializeSimulation(null, data, attackingUnits.iterator().next().getOwner());
+    final IBattleCalculator fastOddsEstimator = new FastOddsEstimator(proData);
+    final AggregateResults fastOddsResults =
+        fastOddsEstimator.calculate(
+            attackingUnits.iterator().next().getOwner(),
+            defendingUnits.iterator().next().getOwner(),
+            territory,
+            attackingUnits,
+            defendingUnits,
+            bombardingUnits,
+            territoryEffects,
+            false,
+            3);
+
+    printResult("FastOdds", fastOddsResults, attackingUnits, defendingUnits, territory, data);
+  }
+
+  private void printResult(
+      final String name,
+      final AggregateResults result,
+      final Collection<Unit> attackingUnits,
+      final Collection<Unit> defendingUnits,
+      final Territory territory,
+      final GameData data) {
+    final DecimalFormat df = new DecimalFormat("0.00");
+    String output = String.format("%-18s", name + ":");
+    output += " Time: " + String.format("%-7.3f", result.getTime() / 1000.0);
+    output += " Win: " + df.format(result.getAttackerWinPercent() * 100.0);
+    output += " AvgRounds: " + String.format("%.1f", result.getAverageBattleRoundsFought());
+    final List<Unit> mainCombatAttackers =
+        CollectionUtils.getMatches(
+            attackingUnits, Matches.unitCanBeInBattle(true, !territory.isWater(), 1, true));
+    final List<Unit> mainCombatDefenders =
+        CollectionUtils.getMatches(
+            defendingUnits, Matches.unitCanBeInBattle(false, !territory.isWater(), 1, true));
+    output +=
+        " Avg Tuv Swing: "
+            + df.format(
+                result.getAverageTuvSwing(
+                    attackingUnits.iterator().next().getOwner(),
+                    mainCombatAttackers,
+                    defendingUnits.iterator().next().getOwner(),
+                    mainCombatDefenders,
+                    data));
+    final Tuple<Double, Double> tuvUnitsLeft =
+        result.getAverageTuvOfUnitsLeftOver(
+            TuvUtils.getCostsForTuv(attackingUnits.iterator().next().getOwner(), data),
+            TuvUtils.getCostsForTuv(defendingUnits.iterator().next().getOwner(), data));
+    output +=
+        " Avg Tuv Units Left: "
+            + df.format(tuvUnitsLeft.getFirst())
+            + ","
+            + df.format(tuvUnitsLeft.getSecond());
+
+    System.out.println(output);
+    System.out.println("Avg Attacking Units: " + result.getAverageAttackingUnitsRemaining());
+    System.out.println("Avg Defending Units: " + result.getAverageDefendingUnitsRemaining());
```

Comment:
System.out is nice for debugging, but in the end you should use a logger instead